samples/template-operator/README.md

```diff
@@ -34,270 +66,511 @@ Additionally, it hides Kubernetes boilerplate code to develop fast and efficient
+        chartPath: "./module-chart",
+        configFlags: types.Flags{
+		  "Namespace":       "redis", // can be omitted if namespace is pre-existing
+		  "CreateNamespace": true, // can be omitted if namespace is pre-existing
+        },
+        setFlags: types.Flags{
+            "nameOverride": "custom-name-override",
+        },
+    }
+    
+    // ManifestResolver represents the chart information for the passed Sample resource.
+    type ManifestResolver struct {
+        chartPath   string
+        setFlags    types.Flags
+        configFlags types.Flags
+    }
+    
+    // Get returns the chart information to be processed.
+    func (m *ManifestResolver) Get(obj types.BaseCustomObject, _ logr.Logger) (types.InstallationSpec, error) {
+        sample, valid := obj.(*operatorv1alpha1.Sample)
+        if !valid {
+            return types.InstallationSpec{},
                 fmt.Errorf("invalid type conversion for %s", client.ObjectKeyFromObject(obj))
-            }
-             return types.InstallationSpec{
-                 ChartPath:   chartPath,
-                 ReleaseName: sample.Spec.ReleaseName,
-                 ChartFlags: types.ChartFlags{
-                    ConfigFlags: types.Flags{
-                        "Namespace":       chartNs,
-                        "CreateNamespace": true,
-                    },
-                    SetFlags: types.Flags{
-                        "nameOverride": nameOverride,
-                    },
-                 },
-            }, nil
-      }
-   ```
+        }
+        return types.InstallationSpec{
+            ChartPath:   m.chartPath,
+            ReleaseName: sample.Spec.Foo,
+            ChartFlags: types.ChartFlags{
+                ConfigFlags: m.configFlags,
+                SetFlags:    m.setFlags,
+            },
+        }, nil
+    }
+   
+    // SampleReconciler reconciles a Sample object
+    type SampleReconciler struct {
+      declarative.ManifestReconciler
+      client.Client
+      Scheme *runtime.Scheme
+    }
    
-4. Run `make generate`, `make manifests` and in the end `make install`, to generate boilerplate code, CRDs and install required resources on your clusterrespectively.
+    // SetupWithManager sets up the controller with the Manager.
+    func (r *SampleReconciler) SetupWithManager(mgr ctrl.Manager) error {
+    if err := r.Inject(
+      mgr, &operatorv1alpha1.Sample{},
+      // Note that now, we need to add it to the Injection so that it's picked up by the declarative reconciler with `declarative.WithManifestResolver(defaultResolver)`
+	  declarative.WithManifestResolver(defaultResolver),
+      declarative.WithCustomResourceLabels(map[string]string{"sampleKey": "sampleValue"}),
+      declarative.WithResourcesReady(true),
+      declarative.WithFinalizer("sample-finalizer"), 
+    ); err != nil {
+      return err
+    }
+    
+    return ctrl.NewControllerManagedBy(mgr).
+	    For(&operatorv1alpha1.Sample{}).
+	    Complete(r)
+    }
```

Comment:
make it collapsible? https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab